### PR TITLE
libpng: Enable intrinsics on x86/SSE2, ppc64/VSX, and all arm/NEON

### LINKED ISSUE #56448: Cannot build for ppc64le
### Godot version

3.4

### System information

Fedora, ppc64le

### Issue description

I tried to build godot on this platform, and I get the following error:
```
[bkeys@great-fox godot]$ scons use_llvm=yes target=release_debug p=linuxbsd
scons: Reading SConscript files ...
Platform "linuxbsd" is still called "x11" in Godot 3.x. Building for platform "x11".
Checking for C header file mntent.h... (cached) yes
scons: done reading SConscript files.
scons: Building targets ...
[ 29%] Compiling ==> thirdparty/embree/common/sys/sysinfo.cpp
In file included from thirdparty/embree/common/sys/sysinfo.cpp:5:
In file included from thirdparty/embree/common/sys/intrinsics.h:15:
In file included from /usr/lib64/clang/13.0.0/include/immintrin.h:17:
/usr/lib64/clang/13.0.0/include/ppc_wrappers/mmintrin.h:31:2: error: "Please read comment above.  Use -DNO_WARN_X86_INTRINSICS to disable this error."
#error                                                                         \
 ^
In file included from thirdparty/embree/common/sys/sysinfo.cpp:5:
In file included from thirdparty/embree/common/sys/intrinsics.h:15:
In file included from /usr/lib64/clang/13.0.0/include/immintrin.h:22:
/usr/lib64/clang/13.0.0/include/ppc_wrappers/xmmintrin.h:31:2: error: "Please read comment above. Use -DNO_WARN_X86_INTRINSICS to disable this error."
#error "Please read comment above. Use -DNO_WARN_X86_INTRINSICS to disable this error."
 ^
In file included from thirdparty/embree/common/sys/sysinfo.cpp:5:
In file included from thirdparty/embree/common/sys/intrinsics.h:15:
In file included from /usr/lib64/clang/13.0.0/include/immintrin.h:22:
In file included from /usr/lib64/clang/13.0.0/include/ppc_wrappers/xmmintrin.h:56:
/usr/lib64/clang/13.0.0/include/ppc_wrappers/mmintrin.h:31:2: error: "Please read comment above.  Use -DNO_WARN_X86_INTRINSICS to disable this error."
#error                                                                         \
 ^
In file included from thirdparty/embree/common/sys/sysinfo.cpp:5:
In file included from thirdparty/embree/common/sys/intrinsics.h:15:
In file included from /usr/lib64/clang/13.0.0/include/immintrin.h:27:
/usr/lib64/clang/13.0.0/include/ppc_wrappers/emmintrin.h:32:2: error: "Please read comment above.  Use -DNO_WARN_X86_INTRINSICS to disable this error."
#error "Please read comment above.  Use -DNO_WARN_X86_INTRINSICS to disable this error."
 ^
In file included from thirdparty/embree/common/sys/sysinfo.cpp:5:
In file included from thirdparty/embree/common/sys/intrinsics.h:15:
In file included from /usr/lib64/clang/13.0.0/include/immintrin.h:27:
In file included from /usr/lib64/clang/13.0.0/include/ppc_wrappers/emmintrin.h:43:
/usr/lib64/clang/13.0.0/include/ppc_wrappers/xmmintrin.h:31:2: error: "Please read comment above. Use -DNO_WARN_X86_INTRINSICS to disable this error."
#error "Please read comment above. Use -DNO_WARN_X86_INTRINSICS to disable this error."
 ^
In file included from thirdparty/embree/common/sys/sysinfo.cpp:5:
In file included from thirdparty/embree/common/sys/intrinsics.h:15:
In file included from /usr/lib64/clang/13.0.0/include/immintrin.h:32:
/usr/lib64/clang/13.0.0/include/ppc_wrappers/pmmintrin.h:35:2: error: "Please read comment above.  Use -DNO_WARN_X86_INTRINSICS to disable this warning."
#error "Please read comment above.  Use -DNO_WARN_X86_INTRINSICS to disable this warning."
 ^
In file included from thirdparty/embree/common/sys/sysinfo.cpp:5:
In file included from thirdparty/embree/common/sys/intrinsics.h:15:
In file included from /usr/lib64/clang/13.0.0/include/immintrin.h:32:
In file included from /usr/lib64/clang/13.0.0/include/ppc_wrappers/pmmintrin.h:44:
/usr/lib64/clang/13.0.0/include/ppc_wrappers/emmintrin.h:32:2: error: "Please read comment above.  Use -DNO_WARN_X86_INTRINSICS to disable this error."
#error "Please read comment above.  Use -DNO_WARN_X86_INTRINSICS to disable this error."
 ^
In file included from thirdparty/embree/common/sys/sysinfo.cpp:5:
In file included from thirdparty/embree/common/sys/intrinsics.h:15:
In file included from /usr/lib64/clang/13.0.0/include/immintrin.h:37:
In file included from /usr/lib64/clang/13.0.0/include/ppc_wrappers/tmmintrin.h:33:
/usr/lib64/clang/13.0.0/include/ppc_wrappers/pmmintrin.h:35:2: error: "Please read comment above.  Use -DNO_WARN_X86_INTRINSICS to disable this warning."
#error "Please read comment above.  Use -DNO_WARN_X86_INTRINSICS to disable this warning."
 ^
In file included from thirdparty/embree/common/sys/sysinfo.cpp:5:
In file included from thirdparty/embree/common/sys/intrinsics.h:15:
In file included from /usr/lib64/clang/13.0.0/include/immintrin.h:42:
/usr/lib64/clang/13.0.0/include/ppc_wrappers/smmintrin.h:25:2: error: "Please read comment above.  Use -DNO_WARN_X86_INTRINSICS to disable this error."
#error                                                                         \
 ^
In file included from thirdparty/embree/common/sys/sysinfo.cpp:5:
In file included from thirdparty/embree/common/sys/intrinsics.h:15:
In file included from /usr/lib64/clang/13.0.0/include/immintrin.h:42:
In file included from /usr/lib64/clang/13.0.0/include/ppc_wrappers/smmintrin.h:35:
/usr/lib64/clang/13.0.0/include/ppc_wrappers/emmintrin.h:32:2: error: "Please read comment above.  Use -DNO_WARN_X86_INTRINSICS to disable this error."
#error "Please read comment above.  Use -DNO_WARN_X86_INTRINSICS to disable this error."
 ^
In file included from thirdparty/embree/common/sys/sysinfo.cpp:5:
In file included from thirdparty/embree/common/sys/intrinsics.h:15:
In file included from /usr/lib64/clang/13.0.0/include/immintrin.h:47:
In file included from /usr/lib64/clang/13.0.0/include/wmmintrin.h:13:
/usr/lib64/clang/13.0.0/include/ppc_wrappers/emmintrin.h:32:2: error: "Please read comment above.  Use -DNO_WARN_X86_INTRINSICS to disable this error."
#error "Please read comment above.  Use -DNO_WARN_X86_INTRINSICS to disable this error."
 ^
In file included from thirdparty/embree/common/sys/sysinfo.cpp:5:
In file included from thirdparty/embree/common/sys/intrinsics.h:15:
In file included from /usr/lib64/clang/13.0.0/include/immintrin.h:47:
In file included from /usr/lib64/clang/13.0.0/include/wmmintrin.h:15:
/usr/lib64/clang/13.0.0/include/__wmmintrin_aes.h:37:19: error: use of undeclared identifier '__builtin_ia32_aesenc128'
  return (__m128i)__builtin_ia32_aesenc128((__v2di)__V, (__v2di)__R);
                  ^
/usr/lib64/clang/13.0.0/include/__wmmintrin_aes.h:57:19: error: use of undeclared identifier '__builtin_ia32_aesenclast128'
  return (__m128i)__builtin_ia32_aesenclast128((__v2di)__V, (__v2di)__R);
                  ^
/usr/lib64/clang/13.0.0/include/__wmmintrin_aes.h:77:19: error: use of undeclared identifier '__builtin_ia32_aesdec128'
  return (__m128i)__builtin_ia32_aesdec128((__v2di)__V, (__v2di)__R);
                  ^
/usr/lib64/clang/13.0.0/include/__wmmintrin_aes.h:97:19: error: use of undeclared identifier '__builtin_ia32_aesdeclast128'
  return (__m128i)__builtin_ia32_aesdeclast128((__v2di)__V, (__v2di)__R);
                  ^
/usr/lib64/clang/13.0.0/include/__wmmintrin_aes.h:114:19: error: use of undeclared identifier '__builtin_ia32_aesimc128'; did you mean '__builtin_asinf128'?
  return (__m128i)__builtin_ia32_aesimc128((__v2di)__V);
                  ^
/usr/lib64/clang/13.0.0/include/__wmmintrin_aes.h:37:19: note: '__builtin_asinf128' declared here
  return (__m128i)__builtin_ia32_aesenc128((__v2di)__V, (__v2di)__R);
                  ^
/usr/lib64/clang/13.0.0/include/__wmmintrin_aes.h:114:44: error: cannot initialize a parameter of type '__float128' with an rvalue of type '__v2di' (vector of 2 'long long' values)
  return (__m128i)__builtin_ia32_aesimc128((__v2di)__V);
                                           ^~~~~~~~~~~
In file included from thirdparty/embree/common/sys/sysinfo.cpp:5:
In file included from thirdparty/embree/common/sys/intrinsics.h:15:
In file included from /usr/lib64/clang/13.0.0/include/immintrin.h:52:
/usr/lib64/clang/13.0.0/include/clflushoptintrin.h:22:3: error: use of undeclared identifier '__builtin_ia32_clflushopt'
  __builtin_ia32_clflushopt(__m);
  ^
In file included from thirdparty/embree/common/sys/sysinfo.cpp:5:
In file included from thirdparty/embree/common/sys/intrinsics.h:15:
In file included from /usr/lib64/clang/13.0.0/include/immintrin.h:57:
/usr/lib64/clang/13.0.0/include/clwbintrin.h:33:3: error: use of undeclared identifier '__builtin_ia32_clwb'
  __builtin_ia32_clwb(__p);
  ^
fatal error: too many errors emitted, stopping now [-ferror-limit=]
20 errors generated.
scons: *** [thirdparty/embree/common/sys/sysinfo.x11.opt.tools.64.llvm.o] Error 1
scons: building terminated because of errors.
[Time elapsed: 00:00:10.933]
```
It looks like it is trying to use x86 specific instructions.

### Steps to reproduce

clone the godot repo, and check into the 3.4 branch, run the following command:
```scons use_llvm=yes target=release_debug p=linuxbsd```

### Minimal reproduction project

_No response_

### PR DESCRIPTION
**Pull request by akien-mga**
We've been enabling NEON optimizations conditionally for Android arm32 only, but recent libpng versions now default to enabling SSE2 for x86 and VSX for ppc64, so we should do the same. The NEON optimizations should also be relevant for arm64 and other non-Android arm platforms, so I'm compiling the files unconditionally.

This will likely not work out of the box and needs heavy testing on all platforms which support either `x86*`, `arm*` and `ppc64`, notably around the weird hack to compile the `.S` file for arm, which apparently didn't work on Windows in the past.

Here's the relevant section of `pngpriv.h` (only compiled with `.c` files, so we don't need those defines in the main env) which handles checking what's enabled at compile time:
https://github.com/godotengine/godot/blob/master/thirdparty/libpng/pngpriv.h#L91-L277

Only `PNG_INTEL_SSE` needs to be passed manually to enable the checks for x86, for other platforms the checks are performed automatically.

This should fix PPC64 build as reported in #56448 (CC @bkeys, could you test?).

Finally, if anyone is interested in checking what this actually optimizes, and do some before/after tests to compare performance, that'd be great!

To do checks, you can use this test build made with the official buildsystem: [downloads.tuxfamily.org/godotengine/testing/4.1-beta%2bpr78325](https://downloads.tuxfamily.org/godotengine/testing/4.1-beta%2bpr78325/)

Should be compared with 4.1 beta 2 or the upcoming 4.1 beta 3 (it's from a commit in the middle of the two).

Fixes #56448.

### CODE REVIEW

File: thirdparty/README.md [hunk 1]
```diff
@@ -314,14 +314,14 @@ Files extracted from upstream source:
 ## libpng
 
 - Upstream: http://libpng.org/pub/png/libpng.html
-- Version: 1.6.38 (0a158f3506502dfa23edfc42790dfaed82efba17, 2022)
+- Version: 1.6.39 (07b8803110da160b158ebfef872627da6c85cbdf, 2022)
```
Comment (akien-mga) [inline thirdparty/README.md:317]: This had just been missed in #70394, we were already on 1.6.39.

File: drivers/png/SCsub [hunk 2]
```diff
@@ -57,6 +47,13 @@ if env["builtin_libpng"]:
         neon_sources.append(env_neon.Object(thirdparty_dir + "/arm/filter_neon.S"))
```
Comment (bruvzg) [inline drivers/png/SCsub:47]: Assembly files probably won't be compatible with MSVC build for ARM Windows, so should have `not env.msvc` checked.

**Comment by bruvzg** [comment]
> iOS fails building because of this pretty fishy part of the setup, which tries to use gcc:

Probably should be:
```python
    compiler_path = "$IOS_TOOLCHAIN_PATH/usr/bin/${ios_triple}"

    ccache_path = os.environ.get("CCACHE")
    if ccache_path is None:
        env["CC"] = compiler_path + "clang"
        env["CXX"] = compiler_path + "clang++"
        env["S_compiler"] = compiler_path + "clang"
    else:
        # there aren't any ccache wrappers available for iOS,
        # to enable caching we need to prepend the path to the ccache binary
        env["CC"] = ccache_path + " " + compiler_path + "clang"
        env["CXX"] = ccache_path + " " + compiler_path + "clang++"
        env["S_compiler"] = ccache_path + " " + compiler_path + "clang"
```

**Comment by bkeys** [comment]
I can't test this out because the source code download isn't available, only binaries and none of the binaries are ppc64le binaries.

**Comment by akien-mga** [comment]
There's a source tarball on the folder I linked, and the source code is also just this PR which you can check out locally.

**Comment by bkeys** [comment]
I checked it out and tried building it and I get the same error I always have
I ran `scons platform=linuxbsd use_llvm=yes target=template_release production=yes`
And the build failed at the link stage:
```
[Initial build] Linking Program bin/godot.linuxbsd.template_release.ppc64.llvm ...
/usr/bin/ld: /tmp/lto-llvm-f0f753.o: in function `png_read_row':
ld-temp.o:(.text.png_read_row+0x19c): undefined reference to `png_init_filter_functions_vsx'
clang-16: error: linker command failed with exit code 1 (use -v to see invocation)
scons: *** [bin/godot.linuxbsd.template_release.ppc64.llvm] Error 1
scons: building terminated because of errors.
```

**Comment by akien-mga** [comment]
Can you confirm that both files in `thirdparty/libpng/powerpc` got compiled? (i.e. they have a matching `.ppc64.o` file)

Does it work without `production=yes` (i.e. with enabling LTO)? To test properly you'd need to clean past build artifacts to make sure it's not wrongly reusing LTO object files.

**Comment by bkeys** [comment]
I'm not sure about the thirdparty files, here is the output of me ls'ing my `thirdparty/libpng` directory:
```
arm                                              pngrio.c
LICENSE                                          pngrio.linuxbsd.template_release.ppc64.llvm.o
png.c                                            pngrtran.c
pngconf.h                                        pngrtran.linuxbsd.template_release.ppc64.llvm.o
pngdebug.h                                       pngrutil.c
pngerror.c                                       pngrutil.linuxbsd.template_release.ppc64.llvm.o
pngerror.linuxbsd.template_release.ppc64.llvm.o  pngset.c
pngget.c                                         pngset.linuxbsd.template_release.ppc64.llvm.o
pngget.linuxbsd.template_release.ppc64.llvm.o    pngstruct.h
png.h                                            pngtrans.c
pnginfo.h                                        pngtrans.linuxbsd.template_release.ppc64.llvm.o
pnglibconf.h                                     pngwio.c
png.linuxbsd.template_release.ppc64.llvm.o       pngwio.linuxbsd.template_release.ppc64.llvm.o
pngmem.c                                         pngwrite.c
pngmem.linuxbsd.template_release.ppc64.llvm.o    pngwrite.linuxbsd.template_release.ppc64.llvm.o
pngpread.c                                       pngwtran.c
pngpread.linuxbsd.template_release.ppc64.llvm.o  pngwtran.linuxbsd.template_release.ppc64.llvm.o
pngpriv.h                                        pngwutil.c
pngread.c                                        pngwutil.linuxbsd.template_release.ppc64.llvm.o
pngread.linuxbsd.template_release.ppc64.llvm.o
```
I still get a failure to link with the following error:
```
[Initial build] Linking Program bin/godot.linuxbsd.template_release.ppc64.llvm ...
/usr/bin/ld: drivers/libdrivers.linuxbsd.template_release.ppc64.llvm.a(pngrutil.linuxbsd.template_release.ppc64.llvm.o): in function `png_read_filter_row':
pngrutil.c:(.text+0x6150): undefined reference to `png_init_filter_functions_vsx'
clang-16: error: linker command failed with exit code 1 (use -v to see invocation)
scons: *** [bin/godot.linuxbsd.template_release.ppc64.llvm] Error 1
scons: building terminated because of errors.
```

**Comment by akien-mga** [comment]
You're missing the `thirdparty/libpng/powerpc` directory added by this PR, so you seem to be compiling the wrong branch.

Try with either this tarball: https://downloads.tuxfamily.org/godotengine/testing/4.1-beta%2bpr78325/godot-4.1-beta.tar.xz

Or the snapshot from my branch: https://github.com/akien-mga/godot/archive/libpng-moar-intrinsics/godot-libpng-moar-intrinsics.zip

**Comment by bkeys** [comment]
It built but now I get this runtime error.

**Comment by akien-mga** [comment]
Yeah it's as fire says. So it's working! Thanks for testing.

You can build `target=editor` if you want to test the editor build.

**Comment by bkeys** [comment]
Yes this pull request seems to have solved the issue; it worked out of the box.

**Comment by bkeys** [comment]
I'm not sure if you already know this, but I can only run a project with the Compatibility rendering:
```
[bkeys@desktop godot-libpng-moar-intrinsics]$ Vulkan API 1.3.246 - Forward Mobile - Using Vulkan Device #0: Unknown - llvmpipe (LLVM 16.0.4, 128 bits)
 
WARNING: Blend file import is enabled in the project settings, but no Blender path is configured in the editor settings. Blend files will not be imported.
     at: _editor_init (modules/gltf/register_types.cpp:73)
ERROR: Caller thread can't call this function in this node (/root). Use call_deferred() or call_thread_group() instead.
   at: propagate_notification (scene/main/node.cpp:2207)

================================================================
handle_crash: Program crashed with signal 11
Engine version: Godot Engine v4.1.beta.custom_build
Dumping the backtrace. Please include this when reporting the bug to the project developer.
[1] linux-vdso64.so.1(__kernel_sigtramp_rt64+0) [0x7fff9b140464] (??:0)
[2] llvm::CmpInst::Create(llvm::Instruction::OtherOps, llvm::CmpInst::Predicate, llvm::Value*, llvm::Value*, llvm::Twine const&, llvm::BasicBlock*) (??:0)
[3] /lib64/libLLVM-16.so(+0x1f01ecc) [0x7fff7cf01ecc] (??:0)
[4] /lib64/libLLVM-16.so(+0x1f01fcc) [0x7fff7cf01fcc] (??:0)
[5] /lib64/libLLVM-16.so(+0x1e8db74) [0x7fff7ce8db74] (??:0)
[6] /lib64/libLLVM-16.so(+0x1e9406c) [0x7fff7ce9406c] (??:0)
[7] llvm::InstCombinePass::run(llvm::Function&, llvm::AnalysisManager<llvm::Function>&) (??:0)
[8] /lib64/libLLVM-16.so(+0x4939cfc) [0x7fff7f939cfc] (??:0)
[9] llvm::PassManager<llvm::Function, llvm::AnalysisManager<llvm::Function>>::run(llvm::Function&, llvm::AnalysisManager<llvm::Function>&) (??:0)
[10] /lib64/libLLVM-16.so(+0x36f389c) [0x7fff7e6f389c] (??:0)
[11] llvm::ModuleToFunctionPassAdaptor::run(llvm::Module&, llvm::AnalysisManager<llvm::Module>&) (??:0)
[12] /lib64/libLLVM-16.so(+0x36f353c) [0x7fff7e6f353c] (??:0)
[13] llvm::PassManager<llvm::Module, llvm::AnalysisManager<llvm::Module>>::run(llvm::Module&, llvm::AnalysisManager<llvm::Module>&) (??:0)
[14] /lib64/libLLVM-16.so(LLVMRunPasses+0x37c) [0x7fff7f9741bc] (??:0)
[15] /usr/lib64/libvulkan_lvp.so(+0x2fa620) [0x7fff830fa620] (??:0)
[16] /usr/lib64/libvulkan_lvp.so(+0x470afc) [0x7fff83270afc] (??:0)
[17] /usr/lib64/libvulkan_lvp.so(+0x475ab0) [0x7fff83275ab0] (??:0)
[18] /usr/lib64/libvulkan_lvp.so(+0x3f8f98) [0x7fff831f8f98] (??:0)
[19] /usr/lib64/libvulkan_lvp.so(+0x3f121c) [0x7fff831f121c] (??:0)
[20] /usr/lib64/libvulkan_lvp.so(+0x3f16d0) [0x7fff831f16d0] (??:0)
[21] /usr/lib64/libvulkan_lvp.so(+0x3f1de8) [0x7fff831f1de8] (??:0)
[22] /usr/lib64/libvulkan_lvp.so(+0x374c80) [0x7fff83174c80] (??:0)
[23] /usr/lib64/libvulkan_lvp.so(+0x2bb330) [0x7fff830bb330] (??:0)
[24] /usr/lib64/libvulkan_lvp.so(+0x2bf4ac) [0x7fff830bf4ac] (??:0)
[25] /usr/lib64/libvulkan_lvp.so(+0x2acc5c) [0x7fff830acc5c] (??:0)
[26] /usr/lib64/libvulkan_lvp.so(+0x10a9c8) [0x7fff82f0a9c8] (??:0)
[27] /usr/lib64/libvulkan_lvp.so(+0x10addc) [0x7fff82f0addc] (??:0)
[28] /usr/lib64/libvulkan_lvp.so(+0xf6de0) [0x7fff82ef6de0] (??:0)
[29] /lib64/libc.so.6(+0xb7c64) [0x7fff9acb7c64] (??:0)
-- END OF BACKTRACE --
================================================================
```
```
Godot Engine v4.1.beta.custom_build - https://godotengine.org
OpenGL API 4.5 (Core Profile) Mesa 23.1.2 - Compatibility - Using Device: Mesa - AMD TURKS (DRM 2.50.0 / 6.2.9-300.fc38.ppc64le, LLVM 16.0.5)
 
ERROR: Transient parent has another exclusive child.
   at: set_visible (scene/main/window.cpp:806)
Editing project: /home/bkeys/Devel/Sandbox
Godot Engine v4.1.beta.custom_build - https://godotengine.org
Vulkan API 1.3.246 - Forward+ - Using Vulkan Device #0: Unknown - llvmpipe (LLVM 16.0.4, 128 bits)
[bkeys@desktop godot-libpng-moar-intrinsics]$  
WARNING: Occlusion culling is disabled at build-time.
     at: _print_warning (./servers/rendering/renderer_scene_occlusion_cull.h:165)
WARNING: Blend file import is enabled in the project settings, but no Blender path is configured in the editor settings. Blend files will not be imported.
     at: _editor_init (modules/gltf/register_types.cpp:73)
ERROR: Caller thread can't call this function in this node (/root). Use call_deferred() or call_thread_group() instead.
   at: propagate_notification (scene/main/node.cpp:2207)

================================================================
handle_crash: Program crashed with signal 11
Engine version: Godot Engine v4.1.beta.custom_build
Dumping the backtrace. Please include this when reporting the bug to the project developer.
[1] linux-vdso64.so.1(__kernel_sigtramp_rt64+0) [0x7fffa2000464] (??:0)
[2] llvm::CmpInst::Create(llvm::Instruction::OtherOps, llvm::CmpInst::Predicate, llvm::Value*, llvm::Value*, llvm::Twine const&, llvm::BasicBlock*) (??:0)
[3] /lib64/libLLVM-16.so(+0x1f01ecc) [0x7fff83d01ecc] (??:0)
[4] /lib64/libLLVM-16.so(+0x1f01fcc) [0x7fff83d01fcc] (??:0)
[5] /lib64/libLLVM-16.so(+0x1e8db74) [0x7fff83c8db74] (??:0)
[6] /lib64/libLLVM-16.so(+0x1e9406c) [0x7fff83c9406c] (??:0)
[7] llvm::InstCombinePass::run(llvm::Function&, llvm::AnalysisManager<llvm::Function>&) (??:0)
[8] /lib64/libLLVM-16.so(+0x4939cfc) [0x7fff86739cfc] (??:0)
[9] llvm::PassManager<llvm::Function, llvm::AnalysisManager<llvm::Function>>::run(llvm::Function&, llvm::AnalysisManager<llvm::Function>&) (??:0)
[10] /lib64/libLLVM-16.so(+0x36f389c) [0x7fff854f389c] (??:0)
[11] llvm::ModuleToFunctionPassAdaptor::run(llvm::Module&, llvm::AnalysisManager<llvm::Module>&) (??:0)
[12] /lib64/libLLVM-16.so(+0x36f353c) [0x7fff854f353c] (??:0)
[13] llvm::PassManager<llvm::Module, llvm::AnalysisManager<llvm::Module>>::run(llvm::Module&, llvm::AnalysisManager<llvm::Module>&) (??:0)
[14] /lib64/libLLVM-16.so(LLVMRunPasses+0x37c) [0x7fff867741bc] (??:0)
[15] /usr/lib64/libvulkan_lvp.so(+0x2fa620) [0x7fff89efa620] (??:0)
[16] /usr/lib64/libvulkan_lvp.so(+0x3a8aa4) [0x7fff89fa8aa4] (??:0)
[17] /usr/lib64/libvulkan_lvp.so(+0x3982e8) [0x7fff89f982e8] (??:0)
[18] /usr/lib64/libvulkan_lvp.so(+0x374e28) [0x7fff89f74e28] (??:0)
[19] /usr/lib64/libvulkan_lvp.so(+0x2bb234) [0x7fff89ebb234] (??:0)
[20] /usr/lib64/libvulkan_lvp.so(+0x2bf4ac) [0x7fff89ebf4ac] (??:0)
[21] /usr/lib64/libvulkan_lvp.so(+0x2acc5c) [0x7fff89eacc5c] (??:0)
[22] /usr/lib64/libvulkan_lvp.so(+0x10a9c8) [0x7fff89d0a9c8] (??:0)
[23] /usr/lib64/libvulkan_lvp.so(+0x10addc) [0x7fff89d0addc] (??:0)
[24] /usr/lib64/libvulkan_lvp.so(+0xf6de0) [0x7fff89cf6de0] (??:0)
[25] /lib64/libc.so.6(+0xb7c64) [0x7fffa1ab7c64] (??:0)
-- END OF BACKTRACE --
================================================================
```

**Comment by bkeys** [comment]
Is there a way you could also backport this change for Godot 3.5.2?

**Comment by aaronfranke** [comment]
@bkeys Patch releases are completely frozen, the best we could possibly do is put it in the `3.5` branch for 3.5.3. But that would only happen if it was first backported to `3.x` which corresponds to the upcoming 3.6 release.

**Comment by bkeys** [comment]
@aaronfranke, well as long as it made it's way into 3 that would be great.

**Comment by akien-mga** [comment]
Yeah I'll probably cherry-pick for 3.6 as fter confirming that this actually makes things faster, or at least does not regress.

The Vulkam crash is unrelated, but you're using Lavapipe (software renderer) which isn't official supported by Godot and has bugs. It might have more bugs on ppc64 as I doubt it's a setup Mesa developers test Vulkan apps on regularly.

**Review by YuriSizov — APPROVED** [review]

**Comment by akien-mga** [comment]
Had a quick look at a potential `3.x` cherrypick, the lack of unified `env["arch"]` in the `3.x` branch makes this pretty difficult to do reliably, so I'll skip it for now.

**Comment by YuriSizov** [comment]
Cherry-picked for 4.1.2.

**Pull request merged**